rust/lance/src/index.rs

```diff
@@ -492,17 +487,38 @@ impl DatasetIndexInternalExt for Dataset {
         let index_file = index_dir.child(INDEX_FILE_NAME);
         let reader: Arc<dyn Reader> = self.object_store.open(&index_file).await?.into();
 
-        let proto = open_index_proto(self, reader.as_ref()).await?;
-        match &proto.implementation {
-            Some(Implementation::VectorIndex(vector_index)) => {
-                let dataset = Arc::new(self.clone());
-                crate::index::vector::open_vector_index(dataset, column, uuid, vector_index, reader)
+        let tailing_bytes = read_last_block(reader.as_ref()).await?;
+        let (major_version, minor_version) = read_version(&tailing_bytes)?;
+
+        // the index file is in lance format since version (0,2)
```

Comment:
We should return an error if the major version is not 0 or the minor version is greater than 2.  This indicates a breaking change and any attempt to read it should fail.